pydantic_ai_slim/pydantic_ai/providers/google_gla.py

```diff
@@ -39,7 +39,13 @@ def __init__(self, api_key: str | None = None, http_client: httpx.AsyncClient |
                 'to use the Google GLA provider.'
             )
 
-        self._client = http_client or cached_async_http_client(provider='google-gla')
-        self._client.base_url = self.base_url
+        if http_client is not None:
+            self._client = http_client
+            if str(http_client.base_url) == '':
+                self._client.base_url = self.base_url
+        else:
+            self._client = cached_async_http_client(provider='google-gla')
+            self._client.base_url = self.base_url
```

Comment:
I think we should add the `base_url` parameter and set it to what is in the property.